x-pack/plugins/security_solution/public/detections/components/take_action_dropdown/index.tsx

```diff
@@ -183,69 +181,91 @@ export const TakeActionDropdown = React.memo(
       [eventFilterActions, exceptionActions, isEvent, actionsData.ruleId]
     );
 
-    const addToCaseProps = useMemo(() => {
-      if (ecsData) {
-        return {
-          event: { data: [], ecs: ecsData, _id: ecsData._id },
-          useInsertTimeline: insertTimelineHook,
-          casePermissions,
-          appId: APP_ID,
-          onClose: afterCaseSelection,
-        };
-      } else {
-        return null;
-      }
-    }, [afterCaseSelection, casePermissions, ecsData, insertTimelineHook]);
+    const addToCaseProps = useMemo(
+      () => ({
+        event: { data: detailsData, ecs: ecsData, _id: ecsData._id },
+        useInsertTimeline: insertTimelineHook,
+        casePermissions,
+        appId: APP_ID,
+        onClose: afterCaseSelection,
+      }),
+      [afterCaseSelection, casePermissions, detailsData, ecsData, insertTimelineHook]
+    );
 
-    const panels = useMemo(() => {
-      if (tGridEnabled) {
-        return [
-          {
-            id: 0,
-            items: [
-              ...alertsActionItems,
-              ...addToCaseActionItem(timelineId),
-              ...hostIsolationAction,
-              ...investigateInTimelineAction,
-            ],
-          },
-          {
-            id: 1,
-            title: CHANGE_ALERT_STATUS,
-            content: <EuiContextMenuPanel size="s" items={actionItems} />,
-          },
-          {
-            id: 2,
-            title: ACTION_ADD_TO_CASE,
-            content: [
-              <>{addToCaseProps && timelinesUi.getAddToExistingCaseButton(addToCaseProps)}</>,
-              <>{addToCaseProps && timelinesUi.getAddToNewCaseButton(addToCaseProps)}</>,
+    const hasWritePermissions = useGetUserCasesPermissions()?.crud ?? false;
+
+    const addToCaseAction = useMemo(
+      () =>
+        [
+          TimelineId.detectionsPage,
+          TimelineId.detectionsRulesDetailsPage,
+          TimelineId.active,
+        ].includes(timelineId as TimelineId) && hasWritePermissions
+          ? {
+              actionItem: [
+                {
+                  name: ACTION_ADD_TO_CASE,
+                  panel: 2,
+                  'data-test-subj': 'attach-alert-to-case-dropdown-button',
+                },
+              ],
+              content: [
+                timelinesUi.getAddToExistingCaseButton(addToCaseProps),
+                timelinesUi.getAddToNewCaseButton(addToCaseProps),
+              ],
+            }
+          : { actionItem: [], content: [] },
+      [addToCaseProps, hasWritePermissions, timelineId, timelinesUi]
+    );
+    const panels = useMemo(
```

Comment:
this actually didn't change that much, just made it a little cleaner and extracted `addToCaseAction` to its own `useMemo`